diff --git a/src/controller/controldevice/controller/mapping/sdl/SDLGyroMapping.cpp b/src/controller/controldevice/controller/mapping/sdl/SDLGyroMapping.cpp
index 59af0b0..f3ffacf 100644
--- a/src/controller/controldevice/controller/mapping/sdl/SDLGyroMapping.cpp
+++ b/src/controller/controldevice/controller/mapping/sdl/SDLGyroMapping.cpp
@@ -55,7 +55,7 @@ void SDLGyroMapping::UpdatePad(float& x, float& y) {
              mPortIndex)) {
         if (!SDL_GameControllerHasSensor(gamepad, SDL_SENSOR_GYRO)) {
 #ifndef __ANDROID__
-            continue
+            continue;
 #endif
         }
 
